### ISSUE TITLE
Cannot use generics with builtin functions like `max`, `min`, etc

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#1234]
## Environment data

- Language Server version: Pylance language server 2020.12.1-pre.1 (pyright ea581b3f)
- OS and version: Mac OS X 11.0.1
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.9 (brew)

## Expected behaviour

It should be possible to use generic types with builtin functions like `max`, `min`, etc. This error appeared with the 2020.12.1-pre.1 (pyright ea581b3f) release.

## Actual behaviour

Pylance cannot handle the generic type:

![Screenshot 2020-12-03 at 10 29 54](https://user-images.githubusercontent.com/849479/100990827-a19ddd00-3552-11eb-917c-682c20a7aeb0.png)

![Screenshot 2020-12-03 at 10 30 18](https://user-images.githubusercontent.com/849479/100990840-a5316400-3552-11eb-9b43-1dd8d628b992.png)

## Logs

```
[Info  - 7:43:16 AM] Pylance language server 2020.12.1-pre.1 (pyright ea581b3f) starting
[Info  - 7:43:16 AM] Server root directory: /Users/dbrattli/.vscode/extensions/ms-python.vscode-pylance-2020.12.1-pre.1/dist
[Error - 7:43:16 AM] stubPath /Users/dbrattli/Developer/Github/Expression/typings is not a valid directory.
[Info  - 9:27:27 AM] No configuration file found.
[Info  - 9:27:27 AM] Setting pythonPath for service "Expression": "/usr/local/opt/python@3.9/bin/python3"
...
```

## Code Snippet / Additional information

```python
import builtins
from typing import Any, Iterable, TypeVar

TSource = TypeVar("TSource")


def max(source: Iterable[TSource]) -> TSource:
    """Returns the greatest of all elements of the sequence,
    compared via `max()`."""

    return builtins.max(source)
```

Changing `TSource` to e.g `str`, `int` or `Any` fixes the type problem, but I think that the generic type should work as well.

**Comment by erictraut** [comment]
I think this behavior is correct. The builtin functions require that the first parameter is type `Iterable[SupportsLessThanT]` where `SupportsLessThanT` is a TypeVar that is bound to the protocol class `SupportsLessThan`.

If you change your code accordingly, it will work:
```python
TSource = TypeVar("TSource", bound=SupportsLessThan)
```

Pylance/Pyright would normally be able to provide a better error message, but it's somewhat obscured by the fact that `max` is an overloaded function, so the best error message that we can offer is "none of the overloads match".

**Comment by dbrattli** [comment]
Thanks, that fixed it! So this is also the same as #1236.
